core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/SnapshotApiServiceSpec.scala

```diff
@@ -154,4 +154,45 @@ class SnapshotApiServiceSpec extends ApiServiceSpec {
       sealRoute(services.snapshotRoutes) ~>
       check { assertResult(StatusCodes.NotFound) {status} }
   }
+
+  it should "return 200 when a user lists all workspaces" in withTestDataApiServicesAndUser(testData.userReader.userEmail.value) { services =>
+    Get(s"${testData.wsName.path}/snapshots", httpJson(
+      EnumerateSnapshotRequestBody(
+        offset = 0,
+        limit = 10
+      )
+    )) ~>
+      sealRoute(services.snapshotRoutes) ~>
+      check {
+        val response = responseAs[DataRepoSnapshotList]
+        assertResult(StatusCodes.OK) {status}
+        assert(response.snapshots.size == 0)
```

Comment:
Ideally, the `MockWorkspaceManagerDAO.references` TrieMap that underlies this response would have some values in it, and this assert would look for size > 0. That would exercise all the json deserialization and class conversions.

Reply:
Good point. I added two `create` calls before the list and check that both snapshots get returned. 
Style q: Is there a better way to organize this than nesting calls in `check` blocks?